asset/asset.go

```diff
@@ -2000,7 +2000,8 @@ func (a *Asset) Decode(r io.Reader) error {
 	if err != nil {
 		return err
 	}
-	return stream.Decode(r)
+
+	return TlvStrictDecode(stream, r, KnownAssetLeafTypes)
```

Comment:
I think to be slightly more conservative, we should put this behind a flag that's disabled for now. This protects us in the off chain that someone is using a modified version of the daemon with their own custom flag, or is using a fresh implementation. 

We could check for this in the wild by running the new decode method on the existing assets in proofs uploaded to our Universe.

Reply:
But wouldn't that break the intended forward compatibility?
Isn't the intention that when we ship `v0.4.2` with this code enabled (e.g. no flag, allow new unknown odd types) that we could then with `v0.5.0` add new fields with odd types, being backward compatible to `v0.4.2`?
With the intention that users can remain on `v0.4.2` without their nodes complaining when syncing new proofs created with `v0.5.0`. If they first have to turn on a flag to allow that, doesn't that kind of defeat the purpose?

Reply:
Related, a useful itest could be that, if a proof is created and then extended with an unknown odd field, checking that a client could successfully store that proof (via local Universe, DB, or flat file), and then return the identical proof to another client.